enarx-keep-sgx-shim/src/start.S

```diff
@@ -60,6 +60,33 @@
     pop     %rbx
 .endm
 
+# Clear the extended CPU state (load synthetic state)
+.macro zerox
+    movq    $~0,                    %rax	    # set mask for xrstor in rax
+    movq    $~0,                    %rdx	    # set mask for xrstor in rdx
+    xrstor  xsave(%rip)                             # clear xtd cpu state with synthetic state
+.endm    
+
+    .section .rodata
+xsave:                          # An initialized synthetic xsave area
+    .align 64
+# Legacy
+    .fill   1, 4, 0x037F        # FCW
+    .fill   5, 4, 0
+    .fill   1, 4, 0x1F80        # MXCSR
+    .fill   1, 4, 0xFFFF        # MXCSR_MASK
+    .fill   60, 8, 0
+ 
+# Header
+    .fill   1, 8, 0             # XSTATE_BV
+    .fill   1, 8, 1 << 63       # XCOMP_BV (compaction mode)
+    .fill   6, 8, 0
+    
+    .section .data
+xsavebuff:                      # An empty buffer to save xsave state
+    .align 64
+    .fill 512, 1, 0
```

Comment:
Since this value is a global, it could only be used by one thread. That works today, but would break as soon as we added a second thread. Instead, `xsavebuff` should just be put on the stack  in `syscall` and loaded from the stack after it is restored during `.Lsyscall`.

Reply:
I've fixed it to do this, in probably an awkward way, because `xrstor` and `xsaveopt` only accept values in memory as their operands, for some reason.